core/runtime.rs

```diff
@@ -805,10 +810,27 @@ impl JsRuntime {
     // Take extensions to avoid double-borrow
     let extensions = std::mem::take(&mut self.extensions_with_js);
     for ext in &extensions {
-      let js_files = ext.init_js();
-      for (filename, source) in js_files {
-        // TODO(@AaronO): use JsRuntime::execute_static() here to move src off heap
-        realm.execute_script(self.v8_isolate(), filename, source)?;
+      {
+        let js_files = ext.init_esm();
+        for (filename, source) in js_files {
+          let id = futures::executor::block_on(self.load_side_module(
+            &ModuleSpecifier::parse(filename).unwrap(),
+            Some(source.to_string()),
+          ))
+          .unwrap();
+          let receiver = self.mod_evaluate(id);
+          futures::executor::block_on(self.run_event_loop(false)).unwrap();
+          let r = futures::executor::block_on(receiver).unwrap();
+          r.unwrap();
```

Comment:
We should rework these unwraps into a block that catches all errors and shows some better error message saying that we were unable to execute the code to be snapshotted